wranglertools/import_data.py

```diff
@@ -154,65 +153,76 @@ def md5(path):
 
 
 def attachment(path):
-    """Create an attachment upload object from a filename and embed the attachment as a data url."""
+    """Create an attachment upload object from a filename and embed the attachment as a data url.
+       NOTE: a url or ftp can be used but path must end in filename with extension that will match
+       the magic detected MIME type of that file and be one of the allowed mime types
+    """
     ftp_attach = False
     if not os.path.isfile(path):
         # if the path does not exist, check if it works as a URL
         if path.startswith("ftp://"):  # grab the file from ftp
             print("\nINFO: Attempting to download file from this url %s" % path)
-            with closing(urllib2.urlopen(path)) as r:
-                file_name = path.split("/")[-1]
-                with open(file_name, 'wb') as f:
-                    shutil.copyfileobj(r, f)
-                    path = file_name
-                    ftp_attach = True
+            try:
+                # import pdb; pdb.set_trace()
+                with closing(urllib2.urlopen(path)) as r:
+                    file_name = path.split("/")[-1]
+                    with open(file_name, 'wb') as f:
+                        shutil.copyfileobj(r, f)
+                        path = file_name
+                        ftp_attach = True
+            except urllib2.URLError as e:
+                raise Exception("\nERROR : FTP fetch for 'attachment' failed - {}".format(e))
```

Comment:
I would recommend defining your own Exception type (and catching it in various tests if it meant to be caught). The below snippet will declare a named exception with no additional behavior. 
```
class MySuperSpecialException(Exception):
    """
    Named exception that you can throw like a normal one
    """
    pass
```